bricks/distributed/vector_clock.h

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include "../../bricks/time/chrono.h"
+
+class DiscreteClocks : public std::vector<uint64_t> {
+ public:
+  using vector<uint64_t>::vector;
+  void inc(int i) { std::vector<uint64_t>::operator[](i)++; }
+  void reset(uint32_t size) { std::vector<uint64_t>::resize(size, 0); }
+};
+
+class ContinuousClocks : public std::vector<std::chrono::microseconds> {
+ public:
+  using vector<std::chrono::microseconds>::vector;
+  void inc(int i) { std::vector<std::chrono::microseconds>::operator[](i) = current::time::Now(); }
+
+  void reset(uint32_t size) {
+    auto now = current::time::Now();
+    std::vector<std::chrono::microseconds>::resize(size, now);
+  }
+};
+
+template <class CLOCK = DiscreteClocks>
+class MergeStrategy {
+ public:
+  std::pair<bool, bool> merge(CLOCK &v1, CLOCK &v2, std::function<bool(CLOCK &v1, CLOCK &v2)> validator) {
```

Comment:
Nit: I was a big fan of `pair<bool, bool>` until I began using more human-readable names. A two-fields `struct` with named fields is just better than `.first` / `.second`. And it's as trivial to construct as `return { lhs, rhs };` FWIW.